spellbook/models/looksrare/ethereum/looksrare_ethereum_events.sql

```diff
@@ -0,0 +1,164 @@
+    ELSE 'Single Item Trade' END AS trade_type,
+    -- Count number of items traded for different trade types and erc standards
+    CASE WHEN agg.name is NULL AND erc.value_unique > 1 THEN erc.value_unique
+        WHEN agg.name is NULL AND erc.value_unique is NULL AND erc.count_erc > 1 THEN erc.count_erc
+        WHEN tokens.standard = 'erc1155' THEN erc.value_unique
+        WHEN tokens.standard = 'erc721' THEN erc.count_erc
+        ELSE (SELECT
+                count(1)::bigint cnt
+            FROM {{ source('erc721_ethereum','evt_transfer') }} erc721
+            WHERE erc721.evt_tx_hash = tx_hash
+            ) +    
+            (SELECT
+                count(1)::bigint cnt
+            FROM {{ source('erc1155_ethereum','evt_transfersingle') }} erc1155
+            WHERE erc1155.evt_tx_hash = tx_hash
+            ) END AS number_of_items,
+    looks_rare.category as trade_category,
+    evt_type,
+    seller,
+    buyer,
+    looks_rare.price / power(10,erc20.decimals) AS amount_original,
+    looks_rare.price AS amount_raw,
+    CASE WHEN looks_rare.currency_contract_original = '0x0000000000000000000000000000000000000000' THEN 'ETH' ELSE erc20.symbol END AS currency_symbol,
+    currency_contract,
+    COALESCE(erc.contract_address, nft_contract_address) AS nft_contract_address,
+    looks_rare.contract_address AS project_contract_address,
+    agg.name AS aggregator_name,
+    agg.contract_address AS aggregator_address,
+    tx_hash,
+    tx.from AS tx_from,
+    tx.to AS tx_to,
+    ROUND((2*(looks_rare.price)/100),7) as platform_fee_amount_raw,
+    ROUND((2*(looks_rare.price / power(10,erc20.decimals))/100),7) platform_fee_amount,
+    ROUND((2*(looks_rare.price / power(10,erc20.decimals) * p.price)/100),7) as  platform_fee_amount_usd,
+    '2' as platform_fee_percentage,
+    royalty_fee as royalty_fee_amount_raw,
+    royalty_fee / power(10,erc20.decimals) as royalty_fee_amount,
+    royalty_fee * p.price/ power(10,erc20.decimals) as royalty_fee_amount_usd,
+    royalty_fee / looks_rare.price * 100 as royalty_fee_percentage,
+    royalty_fee_receive_address,
+    royalty_fee_currency_symbol,
+    tx_hash || '-' ||  token_id || '-' ||  seller || '-' || looks_rare.evt_index::string || '-' || evt_type as unique_trade_id
+FROM looks_rare
+INNER JOIN {{ source('ethereum','transactions') }} tx ON tx_hash = tx.hash
+AND tx.block_time > '2022-01-01'
+LEFT JOIN erc_transfers erc ON erc.evt_tx_hash = tx_hash AND erc.token_id_erc = token_id
+LEFT JOIN {{ ref('tokens_ethereum_nft') }} tokens ON tokens.contract_address =  nft_contract_address
+LEFT JOIN  {{ ref('nft_ethereum_aggregators') }} agg ON agg.contract_address = tx.to
+LEFT JOIN {{ source('prices', 'usd') }} p ON p.minute = date_trunc('minute', looks_rare.block_time)
+    AND p.contract_address = currency_contract
+    AND p.blockchain ='ethereum'
+LEFT JOIN {{ ref('tokens_ethereum_erc20') }} erc20 ON erc20.contract_address = currency_contract
+{% if is_incremental() %}
```

Comment:
my last point is on this incremental logic -- can we move this jinja macro from here to within the looks_rare CTE? it would result in lines 161-164 being copied to after line 38 & line 64 (with edit to table/column name for block time), then removed from 161-164. this would allow the CTE to do the filtering if incremental enabled, then outer query will already have the designated subset you are ultimately filtering to, which simplifies join to eth tx's and prices